/*
 * Copyright 2017-2022 O2 Czech Republic, a.s.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
syntax = "proto3";
package cz.o2.proxima.example.product;
option java_outer_classname = "Product";

/* Price data for product. */
message Price {

  /* Price not including VAT */
  float price = 1;

  /* Price including VAT */
  float priceVat = 2;

}

message Details {

  /* Numerical ID of the product. */
  uint32 id = 1;

  /* Name of the product. */
  string name = 2;

  /* Description of the product. */
  string description  = 3;

}

/* Link to associated category for product. */
message Category {

  /* This might probably be stored in some relational storage. This is not covered by this example. */
  uint32 categoryId = 1;

  // might include additional (denormalized) data

}
